kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/annotations/DiagnosticAnnotationUtil.java

```diff
@@ -132,11 +146,15 @@ private DiagnosticAnnotation createKotlinAnnotation(@NotNull PsiErrorElement psi
     }
     
     @NotNull
-    private DiagnosticAnnotation createKotlinAnnotation(@NotNull Diagnostic diagnostic, @NotNull IFile file) {
+    private DiagnosticAnnotation createKotlinAnnotation(@NotNull Diagnostic diagnostic, @NotNull IFile file) throws BadLocationException {
         TextRange range = diagnostic.getTextRanges().get(0);
-        return new DiagnosticAnnotation(
-                LineEndUtil.convertLfToDocumentOffset(diagnostic.getPsiFile().getText(), 
-                        range.getStartOffset(), EditorUtil.getDocument(file)),
+        
+        IDocument document = EditorUtil.getDocument(file);
+        int offset = LineEndUtil.convertLfToDocumentOffset(diagnostic.getPsiFile().getText(), 
+                range.getStartOffset(), document);
+        
+        return new DiagnosticAnnotation(document.getLineOfOffset(offset),
```

Comment:
Maybe it's worth to handle this exception here only once instead of wrapping several calls into try/catch?